### PR TITLE
Allow the clients to decide if validation of SSL certificates should be skipped

### PR DESCRIPTION
**Pull request by jmkk**
Instead of just assuming that in DEBUG root certificates should not be evaluated allow the callers to decide.
In order to not break existing clients, in DEBUG evaluation still is off, but in release builds you can set the allowUntrustedSSLCertificates flag.

### CODE REVIEW

File: SocketRocket/SRWebSocket.h [hunk 1]
```diff
@@ -41,6 +41,12 @@ extern NSString *const SRWebSocketErrorDomain;
 @property (nonatomic, readonly) SRReadyState readyState;
 @property (nonatomic, readonly, retain) NSURL *url;
 
+// Specifies whether SSL trust chain should NOT be evaluated.
+// By default this flag is set to NO, meaning only secure SSL connections are allowed.
+// For DEBUG builds this flag is ignored, and SSL connections are allowed regardless
+// of the certificate trust configuration
+@property (nonatomic, readwrite) BOOL allowUntrustedSSLCertificates;
```
Comment (martinmroz) [inline SocketRocket/SRWebSocket.h]: Total nit, this should be allowsSelfSignedCertificates (https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingIvarsAndTypes.html) -- in the initializers it should be allows just like you've done.

More importantly though, to stop people from hurting themselves, this property should be marked readonly and the only way to specify it is in the initializer.

File: SocketRocket/SRWebSocket.m [hunk 2]
```diff
@@ -581,21 +586,26 @@ - (void)_initializeStreams;
             [SSLOptions setValue:[NSNumber numberWithBool:NO] forKey:(__bridge id)kCFStreamSSLValidatesCertificateChain];
         }
         
+        BOOL allowUntrustedSSLCertificates = self.allowUntrustedSSLCertificates;
+        
 #if DEBUG
-        [SSLOptions setValue:[NSNumber numberWithBool:NO] forKey:(__bridge id)kCFStreamSSLValidatesCertificateChain];
```
Comment (martinmroz) [inline SocketRocket/SRWebSocket.m]: Friendly note, might be easier to read `[NSNumber numberWithBool:NO]` if it was `@NO`!

File: SocketRocket/SRWebSocket.m [hunk 3]
```diff
@@ -581,21 +586,26 @@ - (void)_initializeStreams;
...
-        NSLog(@"SocketRocket: In debug mode.  Allowing connection to any root cert");
+        allowUntrustedSSLCertificates = YES;
 #endif
+
+        if (allowUntrustedSSLCertificates) {
+            [SSLOptions setValue:[NSNumber numberWithBool:NO] forKey:(__bridge id)kCFStreamSSLValidatesCertificateChain];
```
Comment (martinmroz) [inline SocketRocket/SRWebSocket.m]: Friendly note, might be easier to read `[NSNumber numberWithBool:NO]` if it was `@NO`! Wasn't you I know ^_^

**Comment by martinmroz** [comment]
This looks great and we'd love to take this, just the last feedback on dependency injecting instead of having a mutable property! Thanks!

**Comment by martinmroz** [comment]
Oh, there also appear to be some merge conflicts to resolve!

**Comment by jmkk** [comment]
There @martinmroz, let's see if this is better. It's been a while! Thanks.

**Comment by jmkk** [comment]
@martinmroz ...or do we want to wait another 2.5 years? ;-)

**Comment by dfed** [comment]
It won't be that long, we promise! We're doing sweeps a minimum of monthly to try to clear out our backlog. Look for an answer this Friday 😃. Thanks for bearing with us!

**Comment by kenwigginton** [comment]
Thank you @jmkk !

**Pull request merged**